src/types/src/parser.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright(c) 2006 to 2019 ADLINK Technology Limited and others
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+ * v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+#include <assert.h>
+#include <errno.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+#include "os/os.h"
+
+#define YYSTYPE DDS_TS_PARSER_STYPE
+#define YYLTYPE DDS_TS_PARSER_LTYPE
+
+#define YY_TYPEDEF_YY_SCANNER_T
+typedef void* yyscan_t;
+
+#define YY_NO_UNISTD_H 1 /* to surpress #include <unistd.h> */
+
+#include "parser.h"
+#include "idl.parser.h"
+#include "yy_decl.h"
+#include "idl.lexer.h"
+
+extern void dds_ts_stringify(dds_ts_node_t *root_node, char *buffer, size_t len);
+
+extern int
+dds_ts_parse_file(const char *file, void (*error_func)(int line, int column, const char *msg))
+{
+  int err = 0;
+  FILE *fh;
+
+  assert(file != NULL);
+
+OS_WARNING_MSVC_OFF(4996);
+  fh = fopen(file, "rb");
+OS_WARNING_MSVC_ON(4996);
+
+  if (fh == NULL) {
+    err = errno;
+    if (error_func != 0)
+      error_func(0, 0, "Cannot open file");
+  } else {
+    dds_ts_context_t *context = dds_ts_create_context();
+    if (context == NULL) {
+      if (error_func != 0)
+        error_func(0, 0, "Error: out of memory\n");
+      return 2;
+    }
+    dds_ts_context_set_error_func(context, error_func);
+    yyscan_t scanner;
+    dds_ts_parser_lex_init(&scanner);
+    dds_ts_parser_set_in(fh, scanner);
+    err = dds_ts_parser_parse(scanner, context);
+    if (err == 0) {
+      char buffer[1000];
+      dds_ts_stringify(dds_ts_context_get_root_node(context), buffer, 1000);
+OS_WARNING_MSVC_OFF(4996);
+      printf("Result: '%s'\n", buffer);
```

Comment:
@FransFaase, I take it this is for debugging purposes and will be removed at some point? Please note it with a FIXME and initialize the first element in the buffer to zero. dds_ts_stringify probably always terminates, but since there's no return value, there's no way to check. Also, is it really necessary to turn warnings off for printf? I find that a bit hard to believe.

Reply:
Yes, it is only temporarily there to show some 'result' of the parsing. It is going to be replaced by a call to (one of) the back-end.